test/cypress/integration/manage-team-members/edit_permissions_spec.js

```diff
@@ -0,0 +1,123 @@
+          users: [
+            authenticatedUserStubOpts,
+            userWeAreEditingStubOpts,
+            viewOnlyUserStubOpts,
+            viewAndRefundUserStubOpts
+          ]
+        }
+      },
+      {
+        name: 'getInvitedUsersSuccess',
+        opts: {
+          serviceExternalId: SERVICE_EXTERNAL_ID,
+          invites: [
+            {
+              email: 'invited-admin-user@example.com',
+              role: 'admin'
+            },
+            {
+              email: 'invited-view-only-user@example.com',
+              role: 'view-only'
+            },
+            {
+              email: 'invited-view-and-refund-user@example.com',
+              role: 'view-and-refund'
+            }
+          ]
+        }
+      },
+      {
+        name: 'postUpdateServiceRoleSuccess',
+        opts: {
+          role: 'view-and-refund',
+          external_id: EDITING_USER_ID,
+          serviceExternalId: SERVICE_EXTERNAL_ID
+        }
+      }
+    ])
+  })
+
+  it('should display team members page', () => {
+    cy.setEncryptedCookies(AUTHENTICATED_USER_ID, 1)
+
+    cy.visit(`/service/${SERVICE_EXTERNAL_ID}`)
+
+    cy.get('#team-members-admin-list').find('tr').first().find('td').first().find('a').contains('logged-in-user@example.com (you)')
+    cy.get('#team-members-admin-list').find('tr').eq(1).find('td').first().find('a').contains('other-user@example.com')
+    cy.get('#team-members-view-and-refund-list').find('tr').first().find('td').first().find('a').contains('view-and-refund-user@example.com')
+    cy.get('#team-members-view-only-list').find('tr').first().find('td').first().find('a').contains('view-only-user@example.com')
+
+    cy.get('#invited-team-members-admin-list').find('tr').first().find('td').contains('invited-admin-user@example.com')
+    cy.get('#invited-team-members-view-and-refund-list').find('tr').first().find('td').contains('invited-view-and-refund-user@example.com')
+    cy.get('#invited-team-members-view-only-list').find('tr').first().find('td').contains('invited-view-only-user@example.com')
+  })
+
+  it('should redirect to team member details page', () => {
+    cy.get('a').contains('other-user@example.com').click()
+
+    cy.get('h1').contains('Details for other-user@example.com')
+    cy.get('table').find('tr').first().find('td').first().contains('other-user@example.com')
+    cy.get('table').find('tr').eq(1).find('td').first().contains('Administrator')
+    cy.get('table').find('tr').eq(1).find('td').eq(1).find('a').contains('Edit permissions')
+  })
+
+  it('should redirect to edit user permissions page', () => {
+    cy.get('a').contains('Edit permissions').click()
+    cy.get('#role-admin-input').should('exist').should('have.attr', 'checked')
+  })
+
+  it('should update permission', () => {
+    cy.get('#role-view-and-refund-input').click()
+    cy.get('button').contains('Save changes').click()
+    cy.get('.flash-container--good').contains('Permissions have been updated')
+  })
+})
```

Comment:
should there not be another step where we go back to `/service/${SERVICE_EXTERNAL_ID}` and see the user who's permissions that was just edited appears in the right view-and-refund list?

Reply:
Because the tests are driven using stubs rather than any real interaction with adminusers, this would just check that a user with the view-and-refund permission is in the correct list which we're already checking